README
pyLottoSimu

a programm for the german lottery "pick 6 out of 49" system.

Lotto Generator und Simulator

Make the documentation as .pdf file:
epydoc lotto --pdf

Installieren:

Das Programm läuft auf MacOS, Windows und Linux,
und überall dort wo Python und pyqt installieren lässt!

Das Programm benötigt Python 2.7 dazu. www.python.org/download/ 
und Qt4 für Python 2.7 www.riverbankcomputing.com/software/pyqt/download

Start mit: 
python lotto.pyw

Das Layout (lotto.ui) kann mit den Qt4-Designer bearbeitet werden,
kann dann mit diesem Befehl wieder in Python umgewandelt werden:

cd lotto
 lotto_ui_zu_py.bat

Bedienen:
Der Modus kann in der Menüleiste angewählt werden
Anzahl der Zufallszahlen sind zwischen 1 bis 15 wählbar sowie der
 Zahlenbereich zwischen 20 und 90! 

Modus Zufallsgenerator:
Die Zahlen auswählen und mit 'Zufallsgenerator ein' starten

Modus Simulation:
Die Zahlen auswählen und mit 'Start' die Simulation starten.
Dabei kann mit dem Schieber oben links die Geschwindigkeit verändert werden.

Info und Lizenz:
pyLottoSimu

Copyright (C) <2012-2013> Markus Hackspacher

This file is part of pyLottoSimu.

pyLottoverwaltung is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

pyLottoSimu is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU General Public License
along with pyLottoSimu.  If not, see <http://www.gnu.org/licenses/>.



docs/history.rst
History
=======

.. contents::

Version
-------

The current version is 1.8 from Mai 21, 2018.

Preview next version
^^^^^^^^^^^^^^^^^^^^

The program starts with the last selected lotto system.

Support only Python 3 and Qt5.

planned release first quarter 2019.

Version 1.7
^^^^^^^^^^^

release at June 17, 2018, more code tests integrate.

Version 1.6.1
^^^^^^^^^^^^^

Version 1.6.1 from September 2, 2015.

Version 1.6
^^^^^^^^^^^

The  version 1.6 from June 3, 2015 could be
choose between four lottery systems, the German 6 out of 49 is default,
the Austrian 6 out of 45 with a bonus number,
the Euro 5 million from 50 with two Super numbers 1-11
and as well as the American Powerball Lottery 5 of 59 and a
Bonus number in the range 1 to 35th
Under the hood, the program logic of the surface is no longer dependent
and it will in the next version also use the program without a graphical
Enter surface. So the program structure is clearly arranged
and it may be better to reuse parts of the program in other programs.
Furthermore can be with Python tools create an .exe file::

    python setup.py py2exe

If anyone would like to offer this program as a gimmick on his website,
thereby adjusts the link to the homepage it can do like to feel free.

Version 1.5
^^^^^^^^^^^

The Version 1.5 from February 20, 2014, can be start with Qt4 or QT5.

Version 1.4
^^^^^^^^^^^

The renewals in Version 1.4 of October 3, 2013 from pyLottoSimu is
multilingualism there are now nine languages. And in
Program window there is a major change, it can now
Select whether the number overview window to appear automatically after the draw.


lotto/lotto.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'lotto.ui'
#
# Created: Wed Jan  9 22:53:46 2013
#      by: PyQt4 UI code generator 4.9.3
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    _fromUtf8 = lambda s: s

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName(_fromUtf8("MainWindow"))
        MainWindow.resize(480, 375)
        self.Lottozahlen = QtGui.QWidget(MainWindow)
        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.Lottozahlen.sizePolicy().hasHeightForWidth())
        self.Lottozahlen.setSizePolicy(sizePolicy)
        self.Lottozahlen.setMinimumSize(QtCore.QSize(480, 320))
        self.Lottozahlen.setMaximumSize(QtCore.QSize(480, 320))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.Lottozahlen.setFont(font)
        self.Lottozahlen.setAutoFillBackground(False)
        self.Lottozahlen.setStyleSheet(_fromUtf8("QFrame#Rahmen { border-image: url(:/lottokugel.png)}\n"
"#QLabel {    background-color: rgb(220, 220, 220)}\n"
"QLabel#label_zahl, QLabel#label_zahl_2 {    background-color: none}"))
        self.Lottozahlen.setObjectName(_fromUtf8("Lottozahlen"))
        self.Rahmen = QtGui.QFrame(self.Lottozahlen)
        self.Rahmen.setEnabled(True)
        self.Rahmen.setGeometry(QtCore.QRect(0, 0, 480, 320))
        self.Rahmen.setFrameShape(QtGui.QFrame.StyledPanel)
        self.Rahmen.setFrameShadow(QtGui.QFrame.Raised)
        self.Rahmen.setObjectName(_fromUtf8("Rahmen"))
        self.plainTextEdit = QtGui.QPlainTextEdit(self.Rahmen)
        self.plainTextEdit.setEnabled(True)
        self.plainTextEdit.setGeometry(QtCore.QRect(20, 20, 441, 16))
        self.plainTextEdit.setAutoFillBackground(False)
        self.plainTextEdit.setUndoRedoEnabled(False)
        self.plainTextEdit.setReadOnly(True)
        self.plainTextEdit.setObjectName(_fromUtf8("plainTextEdit"))
        self.label_Lottozahlen = QtGui.QLabel(self.Rahmen)
        self.label_Lottozahlen.setGeometry(QtCore.QRect(20, 0, 441, 21))
        font = QtGui.QFont()
        font.setUnderline(False)
        font.setStrikeOut(False)
        self.label_Lottozahlen.setFont(font)
        self.label_Lottozahlen.setAlignment(QtCore.Qt.AlignCenter)
        self.label_Lottozahlen.setObjectName(_fromUtf8("label_Lottozahlen"))
        self.layoutWidget = QtGui.QWidget(self.Rahmen)
        self.layoutWidget.setGeometry(QtCore.QRect(60, 140, 351, 33))
        self.layoutWidget.setObjectName(_fromUtf8("layoutWidget"))
        self.horizontalLayout = QtGui.QHBoxLayout(self.layoutWidget)
        self.horizontalLayout.setMargin(0)
        self.horizontalLayout.setObjectName(_fromUtf8("horizontalLayout"))
        self.label_3 = QtGui.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setAlignment(QtCore.Qt.AlignCenter)
        self.label_3.setObjectName(_fromUtf8("label_3"))
        self.horizontalLayout.addWidget(self.label_3)
        self.anzahl = QtGui.QSpinBox(self.layoutWidget)
        self.anzahl.setMinimum(1)
        self.anzahl.setMaximum(15)
        self.anzahl.setProperty("value", 6)
        self.anzahl.setObjectName(_fromUtf8("anzahl"))
        self.horizontalLayout.addWidget(self.anzahl)
        self.label_2 = QtGui.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName(_fromUtf8("label_2"))
        self.horizontalLayout.addWidget(self.label_2)
        self.hochste = QtGui.QSpinBox(self.layoutWidget)
        self.hochste.setButtonSymbols(QtGui.QAbstractSpinBox.UpDownArrows)
        self.hochste.setMinimum(20)
        self.hochste.setMaximum(90)
        self.hochste.setSingleStep(1)
        self.hochste.setProperty("value", 49)
        self.hochste.setObjectName(_fromUtf8("hochste"))
        self.horizontalLayout.addWidget(self.hochste)
        self.Zufallsgenerator = QtGui.QPushButton(self.Rahmen)
        self.Zufallsgenerator.setGeometry(QtCore.QRect(150, 260, 181, 31))
        self.Zufallsgenerator.setObjectName(_fromUtf8("Zufallsgenerator"))
        self.AusgfeldLeeren = QtGui.QPushButton(self.Rahmen)
        self.AusgfeldLeeren.setGeometry(QtCore.QRect(160, 180, 161, 25))
        self.AusgfeldLeeren.setObjectName(_fromUtf8("AusgfeldLeeren"))
        self.label_zahl = QtGui.QLabel(self.Rahmen)
        self.label_zahl.setEnabled(True)
        self.label_zahl.setGeometry(QtCore.QRect(170, 20, 121, 91))
        font = QtGui.QFont()
        font.setPointSize(32)
        font.setBold(True)
        font.setWeight(75)
        self.label_zahl.setFont(font)
        self.label_zahl.setAlignment(QtCore.Qt.AlignCenter)
        self.label_zahl.setObjectName(_fromUtf8("label_zahl"))
        self.btn_start = QtGui.QPushButton(self.Rahmen)
        self.btn_start.setGeometry(QtCore.QRect(180, 110, 121, 25))
        self.btn_start.setObjectName(_fromUtf8("btn_start"))
        self.horizontalSlider = QtGui.QSlider(self.Rahmen)
        self.horizontalSlider.setGeometry(QtCore.QRect(20, 50, 91, 20))
        self.horizontalSlider.setMinimum(1)
        self.horizontalSlider.setMaximum(99)
        self.horizontalSlider.setProperty("value", 50)
        self.horizontalSlider.setSliderPosition(50)
        self.horizontalSlider.setOrientation(QtCore.Qt.Horizontal)
        self.horizontalSlider.setInvertedAppearance(True)
        self.horizontalSlider.setInvertedControls(False)
        self.horizontalSlider.setObjectName(_fromUtf8("horizontalSlider"))
        self.label_Geschwindigkeit = QtGui.QLabel(self.Rahmen)
        self.label_Geschwindigkeit.setGeometry(QtCore.QRect(15, 80, 121, 20))
        self.label_Geschwindigkeit.setObjectName(_fromUtf8("label_Geschwindigkeit"))
        self.label_zahl_2 = QtGui.QLabel(self.Rahmen)
        self.label_zahl_2.setEnabled(True)
        self.label_zahl_2.setGeometry(QtCore.QRect(380, 50, 41, 41))
        font = QtGui.QFont()
        font.setPointSize(26)
        font.setBold(False)
        font.setWeight(50)
        self.label_zahl_2.setFont(font)
        self.label_zahl_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_zahl_2.setObjectName(_fromUtf8("label_zahl_2"))
        MainWindow.setCentralWidget(self.Lottozahlen)
        self.menubar = QtGui.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 480, 29))
        self.menubar.setObjectName(_fromUtf8("menubar"))
        self.menuDatei = QtGui.QMenu(self.menubar)
        self.menuDatei.setObjectName(_fromUtf8("menuDatei"))
        self.menuHilfe = QtGui.QMenu(self.menubar)
        self.menuHilfe.setObjectName(_fromUtf8("menuHilfe"))
        self.menuModus = QtGui.QMenu(self.menubar)
        self.menuModus.setObjectName(_fromUtf8("menuModus"))
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtGui.QStatusBar(MainWindow)
        self.statusbar.setObjectName(_fromUtf8("statusbar"))
        MainWindow.setStatusBar(self.statusbar)
        self.actionBeenden = QtGui.QAction(MainWindow)
        self.actionBeenden.setObjectName(_fromUtf8("actionBeenden"))
        self.actionInfo = QtGui.QAction(MainWindow)
        self.actionInfo.setObjectName(_fromUtf8("actionInfo"))
        self.actionZufallsgenerator = QtGui.QAction(MainWindow)
        self.actionZufallsgenerator.setCheckable(True)
        self.actionZufallsgenerator.setChecked(True)
        self.actionZufallsgenerator.setObjectName(_fromUtf8("actionZufallsgenerator"))
        self.actionLottosimulation = QtGui.QAction(MainWindow)
        self.actionLottosimulation.setCheckable(True)
        self.actionLottosimulation.setChecked(True)
        self.actionLottosimulation.setObjectName(_fromUtf8("actionLottosimulation"))
        self.actionHilfe = QtGui.QAction(MainWindow)
        self.actionHilfe.setObjectName(_fromUtf8("actionHilfe"))
        self.menuDatei.addAction(self.actionBeenden)
        self.menuHilfe.addAction(self.actionInfo)
        self.menuModus.addAction(self.actionLottosimulation)
        self.menubar.addAction(self.menuDatei.menuAction())
        self.menubar.addAction(self.menuModus.menuAction())
        self.menubar.addAction(self.menuHilfe.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)
        MainWindow.setTabOrder(self.anzahl, self.hochste)

    def retranslateUi(self, MainWindow):
        MainWindow.setWindowTitle(QtGui.QApplication.translate("MainWindow", "pyLottoSimu", None, QtGui.QApplication.UnicodeUTF8))
        self.Lottozahlen.setWindowTitle(QtGui.QApplication.translate("MainWindow", "Lottozahlen", None, QtGui.QApplication.UnicodeUTF8))
        self.plainTextEdit.setWhatsThis(QtGui.QApplication.translate("MainWindow", "Ausgabefeld", None, QtGui.QApplication.UnicodeUTF8))
        self.label_Lottozahlen.setText(QtGui.QApplication.translate("MainWindow", "Lottozahlen", None, QtGui.QApplication.UnicodeUTF8))
        self.label_3.setText(QtGui.QApplication.translate("MainWindow", "Zahlen:", None, QtGui.QApplication.UnicodeUTF8))
        self.anzahl.setToolTip(QtGui.QApplication.translate("MainWindow", "Anzahl", None, QtGui.QApplication.UnicodeUTF8))
        self.anzahl.setWhatsThis(QtGui.QApplication.translate("MainWindow", "Anzahl", None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("MainWindow", "aus", None, QtGui.QApplication.UnicodeUTF8))
        self.hochste.setToolTip(QtGui.QApplication.translate("MainWindow", "Höchste Zufallszahl", None, QtGui.QApplication.UnicodeUTF8))
        self.hochste.setStatusTip(QtGui.QApplication.translate("MainWindow", "Höchste Zufallszahl", None, QtGui.QApplication.UnicodeUTF8))
        self.hochste.setWhatsThis(QtGui.QApplication.translate("MainWindow", "Höchste Zufallszahl", None, QtGui.QApplication.UnicodeUTF8))
        self.Zufallsgenerator.setToolTip(QtGui.QApplication.translate("MainWindow", "Zufallsgenerator ein", None, QtGui.QApplication.UnicodeUTF8))
        self.Zufallsgenerator.setStatusTip(QtGui.QApplication.translate("MainWindow", "Zufallsgenerator ein", None, QtGui.QApplication.UnicodeUTF8))
        self.Zufallsgenerator.setText(QtGui.QApplication.translate("MainWindow", "Zufallsgenerator ein", None, QtGui.QApplication.UnicodeUTF8))
        self.AusgfeldLeeren.setToolTip(QtGui.QApplication.translate("MainWindow", "Ausgabefeld leeren", None, QtGui.QApplication.UnicodeUTF8))
        self.AusgfeldLeeren.setText(QtGui.QApplication.translate("MainWindow", "Ausgabefeld leeren", None, QtGui.QApplication.UnicodeUTF8))
        self.label_zahl.setText(QtGui.QApplication.translate("MainWindow", "Text", None, QtGui.QApplication.UnicodeUTF8))
        self.btn_start.setToolTip(QtGui.QApplication.translate("MainWindow", "Start", None, QtGui.QApplication.UnicodeUTF8))
        self.btn_start.setStatusTip(QtGui.QApplication.translate("MainWindow", "Start", None, QtGui.QApplication.UnicodeUTF8))
        self.btn_start.setText(QtGui.QApplication.translate("MainWindow", "Start", None, QtGui.QApplication.UnicodeUTF8))
        self.horizontalSlider.setToolTip(QtGui.QApplication.translate("MainWindow", "Speed", None, QtGui.QApplication.UnicodeUTF8))
        self.horizontalSlider.setStatusTip(QtGui.QApplication.translate("MainWindow", "Speed", None, QtGui.QApplication.UnicodeUTF8))
        self.label_Geschwindigkeit.setText(QtGui.QApplication.translate("MainWindow", "Geschwindigkeit", None, QtGui.QApplication.UnicodeUTF8))
        self.label_zahl_2.setText(QtGui.QApplication.translate("MainWindow", "Text", None, QtGui.QApplication.UnicodeUTF8))
        self.menuDatei.setTitle(QtGui.QApplication.translate("MainWindow", "Datei", None, QtGui.QApplication.UnicodeUTF8))
        self.menuHilfe.setTitle(QtGui.QApplication.translate("MainWindow", "Hilfe", None, QtGui.QApplication.UnicodeUTF8))
        self.menuModus.setTitle(QtGui.QApplication.translate("MainWindow", "Modus", None, QtGui.QApplication.UnicodeUTF8))
        self.actionBeenden.setText(QtGui.QApplication.translate("MainWindow", "Beenden", None, QtGui.QApplication.UnicodeUTF8))
        self.actionInfo.setText(QtGui.QApplication.translate("MainWindow", "Info", None, QtGui.QApplication.UnicodeUTF8))
        self.actionZufallsgenerator.setText(QtGui.QApplication.translate("MainWindow", "Zufallsgenerator", None, QtGui.QApplication.UnicodeUTF8))
        self.actionLottosimulation.setText(QtGui.QApplication.translate("MainWindow", "Lottosimulation", None, QtGui.QApplication.UnicodeUTF8))
        self.actionHilfe.setText(QtGui.QApplication.translate("MainWindow", "Hilfe", None, QtGui.QApplication.UnicodeUTF8))

import lottokugeln_rc
